### ISSUE TITLE
Embargo for new version docs

### ISSUE THREAD

**Issue by mortenpi** [JuliaLang/julia#46659]
Right now, when a new version gets tagged, we immediately deploy the docs and make them available to the users (e.g. under the `v1/` URL), even before the release is officially announced. This means that for a while, users are accessing docs for a future version of Julia (especially noticeable for minor releases) and it also leads to the link to the PDF not working (since it gets deployed later; although this would be mitigated by https://github.com/JuliaCI/julia-buildkite/issues/198).

Having the docs on `gh-pages` early is not actually a problem as long as we hold off on updating the symlinks and `versioninfo.js`. So what we could do is have a small file sitting on the [`gh-pages` branch](https://github.com/JuliaLang/docs.julialang.org/tree/gh-pages) that declares an embargo on a particular version of the docs, and make sure that `doc/make.jl` doesn't generate the symlinks for that. Once a release is fully out, that file would then have to be manually removed/updated, to remove the embargo.

**Comment by DilumAluthge** [comment]
Yeah, I think it would be great if we could embargo the new docs until after the Discourse announcement has been posted.

**Comment by mgkuhn** [comment]
The real solution is #34278, i.e. making it easier for REPL users to open the built-in documentation in a browser, rather than assuming that the installed version of Julia, or of any package, matches some web page.
